web-dev/src/components/UserProfile/index.js

```diff
@@ -0,0 +1,40 @@
+import React from "react"
+import UserProfileComponent from "./UserProfileComponent"
+import NavigationTopBar from "../NavigationTopBar/index.js";
+import ProfileComments from "../ProfileComments"
+import profile from "../data/profile.json"
+
+import {Link, useLocation} from "react-router-dom";
+import "../Profile/profile.css"
+
+const UserProfile = () => {
+    const otherUser = useLocation().pathname.split(/[/:]+/)[2];
+    return (
+        <>
+            <div className="d-flex bg-color"
+                 style={{flexDirection:"column"}}>
+                <div >
+                    <NavigationTopBar />
+                </div>
+                <div className="wd-top-profile wd-pad-siding ">
+                    <div className="mb-2">
+                        <span className="fw-bold fs-4">{otherUser}'s Account</span>
+
+                    </div>
+                    <UserProfileComponent profile={profile}/>
+                </div>
+
+                <div className="mt-3 wd-pad-siding">
+                    <p className="fw-bold fs-4">{otherUser}'s Comments</p>
+                    <ProfileComments />
```

Comment:
This is the hardest part. Since I'm not sure how to hide the edit and remove icons for the profile comment. I'm guessing we can do some Secure Content aka Authentication Lock for the icons in the ProfileComments's comment-item.js. I'm not sure. This would need discussion.

Reply:
If this is a different page then these would only show up when currentUser.userName === extractedUser.userName